src/FileWriterTask.cxx

```diff
@@ -55,26 +55,42 @@ void FileWriterTask::add_source(Source &&source) {
 int FileWriterTask::hdf_init(rapidjson::Value const &nexus_structure,
                              rapidjson::Value const &config_file,
                              std::vector<StreamHDFInfo> &stream_hdf_info,
-                             std::vector<hid_t> &groups) {
+                             std::vector<hdf5::node::Group> &groups) {
   filename_full = hdf_filename;
   if (!hdf_output_prefix.empty()) {
     filename_full = hdf_output_prefix + "/" + filename_full;
   }
-  auto x = hdf_file.init(filename_full, nexus_structure, config_file,
-                         stream_hdf_info, groups);
-  if (x) {
+
+  try {
+    hdf_file.init(filename_full, nexus_structure, config_file,
+                  stream_hdf_info, groups);
+  }
+  catch (...)
+  {
     LOG(Sev::Warning,
         "can not initialize hdf file  hdf_output_prefix: {}  hdf_filename: {}",
         hdf_output_prefix, hdf_filename);
-    return x;
+    return -1;
   }
+
   return 0;
 }
 
-int FileWriterTask::hdf_close() { return hdf_file.close(); }
+void FileWriterTask::hdf_close()
+{
+  // nest another exception?
```

Comment:
I think you can remove this comment; doesn't look like there is value in passing the exception up here.